Formula/s/snapcraft.rb

```diff
@@ -99,28 +97,31 @@ class Snapcraft < Formula
   end
 
   resource "craft-cli" do
-    url "https://files.pythonhosted.org/packages/79/f5/1ef63100f984130aefcaf11c896ad1116438d96bb2020ea934d0f02dd285/craft_cli-3.1.2.tar.gz"
-    sha256 "f5d7281f278ddd867fc67e63e48e96333d187005a7569a226434527c3ccfbd2b"
+    url "https://files.pythonhosted.org/packages/35/0a/f1612fe57295b6a69ff256dbd0826b90f70b27dda90988f972fa6f8968cc/craft_cli-3.2.0.tar.gz"
+    sha256 "557345bf1d82e93d6525f35837ddd2b216de4152f5572306ec59d805231d348b"
   end
 
   resource "craft-grammar" do
-    url "https://files.pythonhosted.org/packages/4f/75/f4851b02aacbf77ff581c4e09f338920ed0786cf4846c16ae3a65534b9fd/craft_grammar-2.0.3.tar.gz"
-    sha256 "90bb9a8a2bf97df866798e634ec6d365c3bee01183c2e1a16394282b411b96a7"
+    url "https://files.pythonhosted.org/packages/d4/aa/70b1e455d01268de803711385bd4111596a18a2183baeb18ea40ad857401/craft_grammar-2.2.0.tar.gz"
+    sha256 "e91f65c6c9d0f08a2783a9fcf946ece80888cee3c85f4120df204f08cef1d486"
   end
 
   resource "craft-parts" do
-    url "https://files.pythonhosted.org/packages/90/bd/445ea92cb9f17a288d45b2b349b1469e71b9800b03c3bfce65f99cd4a2d9/craft_parts-2.20.0.tar.gz"
-    sha256 "e1099252509eecfe56b841de16ed2430df6effbd271c9899756e17af66314b55"
+    url "https://files.pythonhosted.org/packages/a8/81/96897a36449258bf58f2d052ba206a5bb486e640f891f2bbd75d09fe4550/craft_parts-2.23.0.tar.gz"
+    sha256 "44fc902a262ba584beebab644c4a25f31b724a4a7358604c11e02298be51cb0b"
   end
 
   resource "craft-platforms" do
     url "https://files.pythonhosted.org/packages/44/78/f2c3ef342c9e9fee0127516aee113a28c487a999d35ce4aa944a58bd5939/craft_platforms-0.10.0.tar.gz"
     sha256 "85b8630c0f7436b0832466c1dba8deb040502fdadc1d225fbed15d1e1e38f729"
   end
 
+  # Temporarily downgrade from 3.1.0 to fix the error: No module named 'pylxd'
+  # It should only required on linux, but the dependency is pulled on macOS
+  # Issue ref: https://github.com/canonical/snapcraft/issues/5781
```

Comment:
just side note, adding the comments in the resource list would affect the dependency updates in the future